p2p/nodestate/nodestate.go

```diff
@@ -802,12 +869,26 @@ func (ns *NodeStateMachine) SetField(n *enode.Node, field Field, value interface
 	}
 
 	state := node.state
-	ns.lock.Unlock()
-	if len(f.subs) > 0 {
-		for _, cb := range f.subs {
-			cb(n, Flags{mask: state, setup: ns.setup}, oldValue, value)
+	callback := func() {
+		if len(f.subs) > 0 {
+			for _, cb := range f.subs {
+				cb(n, Flags{mask: state, setup: ns.setup}, oldValue, value, caller)
+			}
```

Comment:
If `len(f.subs) == 0`, the `range` query would just not execute. So I don't see a need for an explicit check there -- or am I missing something?

Reply:
No, it was probably doing something else earlier and I didn't realize that I can remove the check.